pkgs/tools/admin/google-cloud-sdk/components.nix

```
@@ -0,0 +1,175 @@
+{ stdenv
+, lib
+, google-cloud-sdk
+, system
+, snapshotPath
+, ...
+}:
+
+let
+  # Mapping from GCS component architecture names to Nix archictecture names
+  arches = {
+    x86 = "i686";
+    x86_64 = "x86_64";
+    # TODO arm
+  };
+
+  # Mapping from GCS component operating systems to Nix operating systems
+  oses = {
+    LINUX = "linux";
+    MACOSX = "darwin";
+    WINDOWS = "windows";
+    CYGWIN = "cygwin";
+  };
+
+  # Convert an archicecture + OS to a Nix platform
+  toNixPlatform = arch: os:
+    let
+      arch' = builtins.getAttr arch arches;
+      os' = builtins.getAttr os oses;
+    in
+    if (builtins.hasAttr arch arches && builtins.hasAttr os oses)
+    then "${arch'}-${os'}"
+    else throw "unsupported architecture '${arch}' and/or os '${os}'";
```

```suggestion
    let
      arch' = arches.${arch} or (throw "unsupported architecture '${arch}')
      os' = oses.${os} or (throw "unsupported OS '${os}'");
    in
    "${os'}-${arch'}";
```
What do you think of this implementation?

---

Much simpler, I like it a lot!